OpenFeature/src/main/java/dev/openfeature/sdk/async/Extensions.kt

```diff
@@ -1,33 +1,50 @@
 package dev.openfeature.sdk.async
 
+import dev.openfeature.sdk.FeatureProvider
+import dev.openfeature.sdk.OpenFeatureAPI
 import dev.openfeature.sdk.OpenFeatureClient
-import dev.openfeature.sdk.events.EventHandler
 import dev.openfeature.sdk.events.OpenFeatureEvents
 import dev.openfeature.sdk.events.observe
 import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.onStart
 import kotlinx.coroutines.flow.take
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.suspendCancellableCoroutine
 
```

Comment:
Need to put some proper effort in documenting these API's but I think this can go in a separate PR.